### ISSUE TITLE
minor issues

### ISSUE THREAD

**Issue by DJCrashdummy** [drcheap/WindowSizer#11]
- [x] i suggest to close the addon-menu automatically after an item was clicked
  - most of the times the button and the menu would move anyway.
  - if your window is already at the clicked size, it seems that your click wasn't recognized at all.
  - the *"Current size"* doesn't get updated anyways, so it is even a click more to close and reopen the menu again (if you want to check it).
  - and if you click *"Customize..."* i'm pretty sure nobody needs it, and therefore it must be closed once again manually.
- [ ] double applying for sizes as big as your actual screen or bigger is needed
  if you want to apply a size exactly as big as you actual screen or bigger, after clicking it, the window size is slightly smaller than the actual screen size (i guess to also fit the window manager within the screen). and then you have to apply the size once again to finally get exactly the chosen window size. :thinking:
  **edit:** ok... it actually does not really depend on the screen resp. window size, it is because of exceeding the screen boundaries: this behavior can be avoided even if you have a window smaller than your screen and manually move one edge outside of the actual area of your screen. then resizing via *WindowSizer* works like a charm at the first try as it should.
i'm afraid this is because of the WM, its edge snap or for whatever reason and may be different with a different WM. :expressionless:

**Comment by drcheap** [comment]
I am almost certain that it used to close automatically upon selecting an item, not sure when that behavior changed.  I guess this went unnoticed since the primary usage is via hotkeys.

Thanks for bringing this up, I will try adding an explicit call to close it when selecting a size.

**Comment by drcheap** [comment]
So yep, the not closing thing was essentially a workaround for an issue I was having where closing it caused the preset to not even get applied, making the menu kind of useless.  Good news is that I figured out why that was happening, and was able to get it to apply the preset **and** close!

I decided to make it an option to keep the window open after selecting a preset in case someone out there actually likes that old behavior.  This could be useful if one wanted to click through the various presets one after another until finding the best one.  However, even if you set that preference, it will still auto-close the menu when you choose Customize, because as you pointed out that never makes sense to keep it open.
